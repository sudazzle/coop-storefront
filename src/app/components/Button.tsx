import React from 'react';

type ButtonProps = React.ButtonHTMLAttributes<HTMLButtonElement> & {
  active?: boolean
}

export const Button: React.FC<ButtonProps> = ({ active, ...rest }) => {
  let className = "hover:bg-blue-900 ease-in duration-300 hover:scale-105 hover:border-blue-900 hover:text-white border border-blue-300 rounded-xl py-2 px-4"
  
  if (active === true) {
    className += " bg-blue-900 text-white"
  } else {
    className += " bg-white text-blue-900"
  }

  return (
    <button className={className} {...rest} />
  );
}