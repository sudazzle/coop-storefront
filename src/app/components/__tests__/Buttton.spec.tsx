import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { Button } from '../Button';

describe('Page', () => {
  it('renders a button', () => {
    const { container } = render(<Button>Test</Button>)
    expect(screen.getByRole('button')).toBeInTheDocument();
    expect(screen.getByText('Test')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('active state of button is visible', () => {
    render(<Button active={true}>Test</Button>);
    expect(screen.getByRole('button')).toHaveClass('bg-blue-900 text-white');
  });

  it('inactive state of button is visible', () => {
    render(<Button active={false}>Test</Button>);
    expect(screen.getByRole('button')).toHaveClass('bg-white text-blue-900');
  });

  it('clicking the button calls the onClick handler', () => {
    const onClick = jest.fn();
    render(<Button onClick={onClick}>Test</Button>);
    screen.getByRole('button').click();
    expect(onClick).toHaveBeenCalled();
  });
});