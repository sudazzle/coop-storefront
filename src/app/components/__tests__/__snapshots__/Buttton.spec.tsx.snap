// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page renders a button 1`] = `
<div>
  <button
    class="hover:bg-blue-900 ease-in duration-300 hover:scale-105 hover:border-blue-900 hover:text-white border border-blue-300 rounded-xl py-2 px-4 bg-white text-blue-900"
  >
    Test
  </button>
</div>
`;
